cmd/terramate/cli/cloud.go

```diff
@@ -477,30 +461,91 @@ func (c *cli) detectCloudMetadata() {
 	}
 }
 
-func getGithubCommit(ghClient *github.Client, repo string, commitName string) (*github.Commit, error) {
+func getGithubCommit(ghClient *github.Client, owner, repo, commit string) (*github.RepositoryCommit, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
 	defer cancel()
 
-	commit, err := ghClient.Commit(ctx, repo, commitName)
+	rcommit, _, err := ghClient.Repositories.GetCommit(ctx, owner, repo, commit, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	return commit, nil
+	return rcommit, nil
 }
 
-func getGithubPR(ghClient *github.Client, repo string, commitName string) (github.Pulls, error) {
+func listGithubPRs(ghClient *github.Client, owner, repo, commit string) ([]*github.PullRequest, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
 	defer cancel()
 
-	return ghClient.PullsForCommit(ctx, repo, commitName)
+	opt := &github.ListOptions{PerPage: 100}
+
+	var allPulls []*github.PullRequest
+	for {
+		pulls, resp, err := ghClient.PullRequests.ListPullRequestsWithCommit(ctx, owner, repo, commit, opt)
+		if err != nil {
+			return nil, err
+		}
+
+		allPulls = append(allPulls, pulls...)
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allPulls, nil
+}
+
+func isGithubPRMerged(ghClient *github.Client, owner, repo string, pullNumber int) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
+	defer cancel()
+
+	isMerged, _, _ := ghClient.PullRequests.IsMerged(ctx, owner, repo, pullNumber)
```

Comment:
when looking at the `IsMerged()` implementation, we can see that it can fail for several reasons and then we will set the state PR state to `merged` which is misleading/wrong.

I think that if the `pull.GetState()` gives the status `closed` and this request fails then in this case we must carry this information (*unable to check merged state*) and set the PR status to `closed (failed to detect merged state)` or something like that.
@sixstone-qq what do you think?

Reply:
we could set as the information we have `closed` without setting the `merged` state if unknown too. Both are fine for me

Reply:
I meant that this implementation uses `merged=false` even if the request fails, which is misleading.